CoffeeMachine_Task1/src/bg/coffee_machine/PremiumCoffeeMachine.java

```diff
@@ -0,0 +1,99 @@
+package bg.coffee_machine;
+
+public class PremiumCoffeeMachine implements CoffeeMachine {
+    private PremiumContainer container;
+    private boolean autoRefill = false;
+    private final String[] allowedLucks = {
+            "If at first you don't succeed call it version 1.0.",
+            "Today you will make magic happen!",
+            "Have you tried turning it off and on again?",
+            "Life would be much more easier if you had the source code.",
+    };
+
+    public PremiumCoffeeMachine(boolean autoRefill) {  this(); this.autoRefill = autoRefill; }
+    public PremiumCoffeeMachine() {
+        this.container = new PremiumContainer();
+    }
+
+    @Override
+    public Product brew(Beverage beverage) {
+        if(beverage instanceof Espresso)
+            if(enoughEspresso(beverage)){
```

Comment:
Currently there is a bug in the following implementation, consider the following example:

1. There is not enough resources to make Espresso
2. I ask for espresso
3. The machine will be refilled and then decrease the resources for espresso without actually returning one.